### PR TITLE
default orphan delete option for read only update strategy.

### PR DESCRIPTION
**Pull request by morvencao**
ref: https://issues.redhat.com/browse/ACM-13118

### CODE REVIEW

File: pkg/api/resource_types.go [hunk 1]
```diff
@@ -193,7 +178,29 @@ func EncodeManifest(manifest, deleteOption, updateStrategy map[string]interface{
...
+	}
+
+	// set default delete option to Orphan if update strategy is ReadOnly
+	if upStrategy.Type == workv1.UpdateStrategyTypeReadOnly {
+		delOption = &workv1.DeleteOption{
+			PropagationPolicy: workv1.DeletePropagationPolicyTypeOrphan,
+		}
+	}
+
+	if len(deleteOption) != 0 {
```
Comment (clyang82) [inline pkg/api/resource_types.go:195]: that means even if the update_strategy is readonly, the delete_option still can be configured. the requirement is if the update_strategy is readonly, the delete_option is orphan. and cannot be configured.

Comment (morvencao) [reply pkg/api/resource_types.go]: updated this to force deletio option to `Orphan` when update strategy is `ReadOnly`.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**